### PR TITLE
[SiVal, i2c]  Adds the override test

### PR DESCRIPTION
**Pull request by engdoreis**
This PR is easier to review each commit separately.

Fix https://github.com/lowRISC/opentitan/issues/19837

### CODE REVIEW

File: sw/host/opentitanlib/src/test_utils/bitbanging/i2c.rs [hunk 1]
```diff
@@ -202,38 +196,41 @@ pub mod decoder {
     }
 
     impl<const SDA: u8, const SCL: u8> Decoder<SDA, SCL> {
-        /// Returns a sample when a fall clock edge happen.
-        /// The returned sample is taken *before* the falling edge, when SCL is high.
+        /// Loops until the clk transitions to low.
+        /// Returns a symbol (Start|Stop) in case the sda transitions while the clk is high.
         /// The caller must make sure that the clock was high in the previous sample.
-        fn sample_on_fall_edge<I>(samples: &mut Peekable<I>) -> Result<Option<Sample<SDA, SCL>>>
+        fn sample_on_fall_edge<I>(samples: &mut Peekable<I>) -> Result<Option<Symbol>>
```
Comment (nbdd0121) [inline sw/host/opentitanlib/src/test_utils/bitbanging/i2c.rs:202]: I think this doesn't have to be peekable anymore?

File: sw/host/tests/chip/i2c_host_override/src/main.rs [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+) -> Result<()> {
+    let uart = transport.uart("console")?;
+    let gpio_bitbanging = transport.gpio_bitbanging()?;
+
+    const SAMPLES: usize = 5000;
+    let mut samples = vec![0x00; SAMPLES];
+    let output = &[0x03; SAMPLES];
+    let mut waveform = [BitbangEntry::Both(output, &mut samples)];
+
+    let _ = UartConsole::wait_for(
```
Comment (nbdd0121) [inline sw/host/tests/chip/i2c_host_override/src/main.rs:55]:
```suggestion
    UartConsole::wait_for(
```

File: sw/host/tests/chip/i2c_host_override/src/main.rs [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+        &*uart,
+        r".*SiVal: waiting for commands.*?[^\r\n]*",
+        opts.timeout,
+    )?;
+    MemWriteReq::execute(&*uart, backdoor_start_addr, &[1 as u8])?;
+
+    let gpio_pins = transport.gpio_pins(&["IOA7", "IOA8"].map(|s| s.to_string()))?;
```
Comment (nbdd0121) [inline sw/host/tests/chip/i2c_host_override/src/main.rs:62]: nit:
```suggestion
    let gpio_pins = transport.gpio_pins(&["IOA7", "IOA8"].map(|s| s.to_owned()))?;
```

File: sw/host/tests/chip/i2c_host_override/src/main.rs [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+        test_utils::bitbanging::i2c::decoder::Decoder::<0, 1> { buffer: [0; 256] };
+    let decoded = i2c_bitbang_decoder.run(samples)?;
+
+    transport.apply_default_configuration(None)?;
+
+    assert_eq!(
+        vec![
+            test_utils::bitbanging::i2c::decoder::Transfer::Start,
+            test_utils::bitbanging::i2c::decoder::Transfer::Addr {
+                addr: data>>1,
```
Comment (nbdd0121) [inline sw/host/tests/chip/i2c_host_override/src/main.rs:86]:
```suggestion
                addr: data >> 1,
```

File: sw/host/tests/chip/i2c_target/src/main.rs [hunk 5]
```diff
@@ -295,13 +296,25 @@ fn main() -> Result<()> {
...
         execute_test!(test_write_transaction, &opts, &transport, 0x33);
         execute_test!(test_write_transaction_slow, &opts, &transport, 0x33);
-        execute_test!(test_write_read_repeated_start, &opts, &transport, 0x70);
     }
     execute_test!(test_wakeup_normal_sleep, &opts, &transport, 0x33);
     execute_test!(test_wakeup_deep_sleep, &opts, &transport, 0x33, 0);
+
+    transport.pin_strapping("RESET")?.apply()?;
+    transport.pin_strapping("RESET")?.remove()?;
+    let _ = UartConsole::wait_for(&*uart, r"Running [^\r\n]*", opts.timeout)?;
```
Comment (nbdd0121) [inline sw/host/tests/chip/i2c_target/src/main.rs:308]:
```suggestion
    UartConsole::wait_for(&*uart, r"Running [^\r\n]*", opts.timeout)?;
```

File: sw/host/opentitanlib/src/test_utils/bitbanging/i2c.rs (hunk 1 above)
Comment (engdoreis) [reply sw/host/opentitanlib/src/test_utils/bitbanging/i2c.rs:202]: It does because of the function `sample_on_raise_edge`

Comment (nbdd0121) [reply sw/host/opentitanlib/src/test_utils/bitbanging/i2c.rs:202]: A peekable iterator is still an iterator so `&mut I` should work?

**Review by nbdd0121 — APPROVED** [review]

**Review by jwnrt — APPROVED** [review]

**Review by nbdd0121 — APPROVED** [review]

**Comment by engdoreis** [comment]
The CI is not related.

**Pull request merged**